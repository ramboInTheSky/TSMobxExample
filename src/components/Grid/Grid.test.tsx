import React from 'react'
import { shallow, mount } from 'enzyme'

import { Grid, GridProps } from '.'

type Dummy = { firstName: string; lastName: string }

describe('<Grid />', () => {
	let props: GridProps<Dummy>
	beforeEach(() => {
		props = {
			loading: false,
			onRowClicked: jest.fn(()=>{}),
			columnDefs: [
				{
					headerName: 'First Name',
					field: 'firstName',
				},
				{
					headerName: 'Last Name',
					field: 'lastName',
				},
			],
			rowData: [
				{
					firstName: 'Bruce',
					lastName: 'Wayne',
				},
				{
					firstName: 'Bruce',
					lastName: 'Wayne',
				},
			],
		}
	})

	it('renders correctly', () => {
		const component = <Grid {...props} />
		expect(shallow(component)).toMatchSnapshot()
	})

	it('renders when there is no data', () => {
		const component = <Grid {...props} rowData={[]} />
		expect(shallow(component)).toMatchSnapshot()
	})

	it('renders when is loading', () => {
		const component = <Grid {...props} loading={true} />
		expect(shallow(component)).toMatchSnapshot()
	})

})
