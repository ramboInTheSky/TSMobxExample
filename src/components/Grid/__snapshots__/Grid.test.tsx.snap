// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> renders correctly 1`] = `
<Boundary>
  <Styled(WithStyles(Paper))
    className="ag-theme-material"
  >
    <CSSTransitionGroup
      transitionAppear={true}
      transitionAppearTimeout={200}
      transitionEnter={true}
      transitionEnterTimeout={200}
      transitionLeave={true}
      transitionLeaveTimeout={200}
      transitionName="fade"
    >
      <AgGridReact
        animateRows={true}
        columnDefs={
          Array [
            Object {
              "field": "firstName",
              "headerName": "First Name",
            },
            Object {
              "field": "lastName",
              "headerName": "Last Name",
            },
          ]
        }
        defaultColDef={
          Object {
            "cellStyle": Object {
              "textAlign": "left",
            },
            "editable": false,
            "filter": "agTextColumnFilter",
            "minWidth": 100,
            "resizable": true,
            "sortable": true,
            "suppressMenu": true,
          }
        }
        onGridReady={[Function]}
        onRowClicked={[MockFunction]}
        reactNext={true}
        rowData={
          Array [
            Object {
              "firstName": "Bruce",
              "lastName": "Wayne",
            },
            Object {
              "firstName": "Bruce",
              "lastName": "Wayne",
            },
          ]
        }
        rowSelection="single"
        suppressCellSelection={true}
      />
    </CSSTransitionGroup>
  </Styled(WithStyles(Paper))>
</Boundary>
`;

exports[`<Grid /> renders when is loading 1`] = `
<Boundary>
  <Styled(WithStyles(Paper))
    className="ag-theme-material"
  >
    <CSSTransitionGroup
      transitionAppear={true}
      transitionAppearTimeout={200}
      transitionEnter={true}
      transitionEnterTimeout={200}
      transitionLeave={true}
      transitionLeaveTimeout={200}
      transitionName="fade"
    />
  </Styled(WithStyles(Paper))>
</Boundary>
`;

exports[`<Grid /> renders when there is no data 1`] = `
<Boundary>
  <Styled(WithStyles(Paper))
    className="ag-theme-material"
  >
    <CSSTransitionGroup
      transitionAppear={true}
      transitionAppearTimeout={200}
      transitionEnter={true}
      transitionEnterTimeout={200}
      transitionLeave={true}
      transitionLeaveTimeout={200}
      transitionName="fade"
    >
      <AgGridReact
        animateRows={true}
        columnDefs={
          Array [
            Object {
              "field": "firstName",
              "headerName": "First Name",
            },
            Object {
              "field": "lastName",
              "headerName": "Last Name",
            },
          ]
        }
        defaultColDef={
          Object {
            "cellStyle": Object {
              "textAlign": "left",
            },
            "editable": false,
            "filter": "agTextColumnFilter",
            "minWidth": 100,
            "resizable": true,
            "sortable": true,
            "suppressMenu": true,
          }
        }
        onGridReady={[Function]}
        onRowClicked={[MockFunction]}
        reactNext={true}
        rowData={Array []}
        rowSelection="single"
        suppressCellSelection={true}
      />
    </CSSTransitionGroup>
  </Styled(WithStyles(Paper))>
</Boundary>
`;
